Makefile

```diff
@@ -33,7 +33,15 @@ export CGO_ENABLE=0
 # OUT is the directory where dist artifacts and temp files will be created.
 OUT=${GO_TOP}/out
 
-GO ?= go
+# It's more concise to use GO?=$(shell which go)
+# but the following approach uses a more efficient "simply expanded" :=
+# variable instead of a "recursively expanded" =
+ifeq ($(origin GO), undefined)
```

Comment:
what is origin? and where are we chekcing for the go version? 
Also a dumb question: why is the simple Go ?= go insufficient? Most environments where we build and do release has go in its path. And people who don't have it are inevitably going to customize the build (aka experts).

Reply:
"origin" is asking make to report the source of the variable (e.g., environment, command-line, or in the makefile). So, the statement is asking make if the variable is undefined.  It's more concise to use "GO?=$(shell which go)", though ?= uses the slower = (expand on each use, i.e., late-binding) vs := (non-expanded, i.e., early binding), and based on my earlier testing I'm getting a bit paranoid now about how eager make seems to be about executing $shell statements.

Using "GO?=go" defeats the detection of whether go is installed or not.  The intent is to catch the case where somebody either hasn't set up go or has installed it improperly.  It's basically trying to have a nicer error message that whatever else will break next because go isn't installed.

The go version check is handled via "init: check-go-version"  I think all of the go targets have a dependency on "depend" which depends on "init", so having "init" depend on "check-go-version" should cause a version check to be performed before any go targets are built.

Reply:
Can you stick this as a comment (paraphrased) so that someone else doesn't happily remove these ?